textproto/multipart.go

```diff
@@ -372,13 +372,13 @@ func (w *MultipartWriter) Boundary() string {
 // SetBoundary must be called before any parts are created, may only
 // contain certain ASCII characters, and must be non-empty and
 // at most 70 bytes long.
-func (w *MultipartWriter) SetBoundary(boundary string) error {
+func (w *MultipartWriter) SetBoundary(boundary string) (error, error) {
```

Comment:
if you just need to distinguish between two types of errors create them as package constant

https://smyrman.medium.com/writing-constant-errors-with-go-1-13-10c4191617